src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -180,11 +181,16 @@ private static String makeDumpableClassName(String className) {
             if (ctr == null)  ctr = 0;
             DUMP_CLASS_FILES_COUNTERS.put(className, ctr+1);
         }
-        String sfx = ctr.toString();
-        while (sfx.length() < 3)
-            sfx = "0" + sfx;
-        className += sfx;
-        return className;
+
+        var buf = new StringBuilder(className.length() + 3).append(className);
+        int ctrVal = ctr;
+        if (ctrVal < 10) {
+            buf.repeat('0', 2);
+        } else if (ctrVal < 100) {
+            buf.append('0');
+        }
+        buf.append(ctrVal);
```

Comment:
Should we write like:
```java
int digitCount = DecimalDigits.stringSize(ctr);
var buf = new StringBuilder(className.length() + Math.max(3, digitCount)).append(className);
if (digitCount < 3)
    buf.repeat('0', 3 - digitCount);
buf.append(ctr);
```

Reply:
DecimalDigits.stringSize has an overhead, so this may be slower